conference.js

```diff
@@ -2351,6 +2355,18 @@ export default {
             promises.push(setAudioOutputPromise);
         }
 
+        // Handles the use case when the default device is changed (we are always stopping the streams because it's
+        // simpler):
+        // If the default device is changed we need to first stop the local streams and then call GUM. Otherwise GUM
+        // will return a stream using the old default device.
+        if (requestedInput.audio && this.localAudio) {
+            this.localAudio.stopStream();
+        }
+
+        if (requestedInput.video && this.localVideo) {
+            this.localVideo.stopStream();
+        }
```

Comment:
I want to say not stopping the stream first was some kind of optimization, maybe to do with SDP thrashing or just UX. I recall during initial device selection modal creation that stopping the stream first at the time was not desired maybe because maybe at the time this bug didn't exist. Have you confirmed with maybe Damyan that this doesn't have negative effects? I assume whatever effects there are aren't bad but it'd be nice to know why the logic was written the way it was.

Reply:
I will confirm! I was also thinking that we don't do offer/answer when we change devices. But when I tested it turns out we are doing offer/answer anyway...

Reply:
@virtuacoplenny , @damencho said that he doesn't have opinion on the subject. 

But I think we should be good here because we are only stopping the stream and even not removing it from the conference. For the default devices there isn't much we can do - if we don't stop the stream we'll just use the wrong device.